locator/token_metadata.hh

```diff
@@ -45,8 +45,7 @@ class token_metadata;
 class tablet_metadata;
 
 struct host_id_or_endpoint {
-    host_id id;
-    gms::inet_address endpoint;
+    std::variant<host_id, gms::inet_address> _value;
```

Comment:
This looks fishy. Do we have a case where we know the host id, and a separate case where we only know the IP address?

Can't we only store the host id?

Reply:
Another way to look at it: can we find the host_id during construction, and store only that?

Reply:
it represents the values we get from the config option that can be either ip addresses or host_id:s, both we parse from strings that aren't guaranteed to represent valid nodes when they are parsed.
Here we "resolve" them by looking for nodes identified by these addresses.  This just holds the parsed options.

Reply:
BTW we need to deprecate IP address passing.

Reply:
> BTW we need to deprecate IP address passing.

Yes.  (out of scope for this PR)